cmd/limactl/editflags/editflags.go

```diff
@@ -0,0 +1,222 @@
+package editflags
+
+import (
+	"fmt"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/pbnjay/memory"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+)
+
+// RegisterEdit registers flags related to in-place YAML modification, for `limactl edit`.
+func RegisterEdit(cmd *cobra.Command) {
+	registerEdit(cmd, "")
+}
+
+func registerEdit(cmd *cobra.Command, commentPrefix string) {
+	flags := cmd.Flags()
+
+	flags.Int("cpus", 0, commentPrefix+"number of CPUs") // Similar to colima's --cpu, but the flag name is slightly different (cpu vs cpus)
+	_ = cmd.RegisterFlagCompletionFunc("cpus", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		var res []string
+		for i := runtime.NumCPU(); i >= 1; i /= 2 {
+			res = append(res, strconv.Itoa(i))
+		}
```

Comment:
This doesn't work well when the number of CPUs is not a power of 2. On a machine with 20 CPUs I get this:

```
jan@studio ~ % limactl create --cpus
1   10  2   20  5
```

I would have preferred to get `1 2 4 8 16 20` as choices instead of `1 2 5 10 20`.

I guess there is nothing we can do to sort completions numerically instead of alphabetically, or is there?

Reply:
Fixed, but not sortable

Reply:
The same will also be true for some memory configurations; there are M2 chips with 24GB, 96GB, and 192GB of RAM (and others that are proper powers of 2).